### PR TITLE
typing(Thread): improve

### PR DESCRIPTION
**Pull request by Victorsitou**
## Summary.
https://canary.discord.com/channels/808030843078836254/847513235536216085/1104843888205770752

## Checklist

- [ ] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `pdm lint`
    - [x] I have type-checked the code by running `pdm pyright`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by Skelmis** [comment]
Link provided isn't viewable to us regular folk, care to tl;dr?

**Review by Skelmis — APPROVED** [review]

**Comment by Victorsitou** [comment]
> Link provided isn't viewable to us regular folk, care to tl;dr?

tl;dr: basically this
```py
if not isinstance(ctx.channel, DMChannel): # DMChannel does not have name
   name = ctx.channel.name # Pylance(reportUnknownMemberType): Type of "name" is unknown, str | Unknown.

File: disnake/threads.py [hunk 1]
```diff
@@ -208,22 +208,24 @@ def _from_data(self, data: ThreadPayload) -> None:
...
+            self.name: str = data["name"]
         except KeyError:
             pass
 
-        self.slowmode_delay = data.get("rate_limit_per_user", 0)
-        self._flags = data.get("flags", 0)
-        self._applied_tags = list(map(int, data.get("applied_tags", [])))
+        self.slowmode_delay: int = data.get("rate_limit_per_user", 0)
+        self._flags: int = data.get("flags", 0)
+        self._applied_tags: List[int] = list(map(int, data.get("applied_tags", [])))
```
Comment (shiftinv) [inline disnake/threads.py:228]: These attributes already have annotations in `__init__`, there isn't really a need to redeclare the types here

Comment (Victorsitou) [reply disnake/threads.py:228]: I see

Comment (Victorsitou) [reply disnake/threads.py]: 1ea5d74973f637f47ce4a6bb8f2a3c80af1f456d
